### DISCUSSION TITLE
High Security Issue everybody should be aware about

### DISCUSSION THREAD

**Discussion by 4thFourthHorseman** [Chia-Network/chia-blockchain#3363] [General]
Who has a clue how the key management works? I know, I can create a private key, but does the private has to be on my online-computer all the time? Lets take ETH for example: when I mine ETH, I only expose my public wallet address and keep my private key private, usually on a usb stick or an offline computer, which is not connected to the internet. Now, if you ask "why?", or "isn't my private key in the contol of an exchange?".  Yes, you expose your key contol to the the exange, If you open an account. You give up your control over your private key to the exchange. So usually this is something you don't want to do. You could for example create your own private key / Public wallet address and keep the private key safe and mine to your wallet address. This is how it works with ETH. Why is there no documentation about this for Chia?

The question I want to ask is: is it necessery that my private key is stored on my computer while I only create plots and farm or can I save it to a safer location and remove it from my online-computer where I create and transfer plots?

**Comment by cccat6** [comment]
You don't need a private key on the computer to plotting. Plotting only needs your public key. The reason why you have this concern because you are using GUI and running the full node. You can use CLI to run just a part of the function on the computer. If you don't know how to use CLI, check https://github.com/Chia-Network/chia-blockchain/wiki/CLI-Commands-Reference
For architecture you can see here https://github.com/Chia-Network/chia-blockchain/wiki/Network-Architecture
